test/TestUtilities/Test.Utility/TestDirectory.cs

```diff
@@ -93,11 +93,45 @@ public static TestDirectory Create(string root)
             var path = PathIO.Combine(parentPath, Guid.NewGuid().ToString().Split('-')[0]);
             Directory.CreateDirectory(path);
 
+            // Need to clear package source mapping instruction from main repo.
+            CreateFile(parentPath, "nuget.config",
```

Comment:
There're so many test only use `TestDirectory.Create()` to create test context without using `SimpleTestPathContext`, replacing all of them `TestDirectory.Create()` with `SimpleTestPathContext` would be very time consuming and big PR.

Reply:
I get that. It can be a different PR as a prereq to this.

I don't think TestDirectory should have anything to do with configuration.

cc @NuGet/nuget-client 

I would love to hear thoughts.

Reply:
Individual tests use `TestDirectory` do affected by nuget.config on top folder, that's why I added this.

Reply:
> Individual tests use TestDirectory do affected by nuget.config on top folder, that's why I added this

I understand that. 

I'm saying those tests shouldn't use TestDirectory. 

TestDirectory does not need to know about configurations such as this. 

Imo, this a hacky/convenience solution, not a well structured one.